RFS/src/main/java/com/rfs/cms/CmsEntry.java

```diff
@@ -1,11 +1,64 @@
 package com.rfs.cms;
 
+
 import com.rfs.common.RfsException;
 
 public class CmsEntry {
+    public static enum EntryType {
+        SNAPSHOT,
+        METADATA,
+        INDEX,
+        INDEX_WORK_ITEM,
+    }
+
     public abstract static class Base {
         protected Base() {}
+        
+        @Override
         public abstract String toString();
+    
+        @Override
+        public boolean equals(Object obj) {
+            if (this == obj) {
+                return true;
+            }
+            if (obj == null || !(obj instanceof Base)) {
+                return false;
+            }
+            Base other = (Base) obj;
+            return this.toString().equals(other.toString());
+        }
+    
+        @Override
+        public int hashCode() {
+            return this.toString().hashCode();
+        }
+    }
+
+    /*
+     * Provides a base class for leasable entry types.  Doesn't allow for customization of lease mechanics, but it's 
+     * unclear how to achieve that in Java given the constraints around static methods.
+    */
+    public abstract static class Leasable extends Base {
```

Comment:
Per discussion: Maybe the lease entry itself belongs in here, or maybe we can use a factory to set it.